canvas_modules/common-canvas/__tests__/common-properties/controls/toggle-test.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017-2022 Elyra Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import Toggle from "../../../src/common-properties/controls/toggle";
+import { mount } from "enzyme";
+import { expect } from "chai";
+import Controller from "../../../src/common-properties/properties-controller";
+import propertyUtils from "../../_utils_/property-utils";
+import toggleParamDef from "../../test_resources/paramDefs/toggle_paramDef.json";
+
+const controller = new Controller();
+
+const control = {
+	"name": "toggle",
+	"label": {
+		"text": "Toggle"
+	},
+	"values": [
+		"On",
+		"Off"
+	],
+};
+
+propertyUtils.setControls(controller, [control]);
+
+const propertyId = { name: "toggle" };
+
+describe("toggle renders correctly", () => {
+
+	it("toggle props should have been defined", () => {
+		const wrapper = mount(
+			<Toggle
+				store={controller.getStore()}
+				control={control}
+				controller={controller}
+				propertyId={propertyId}
+			/>
+		);
+		expect(wrapper.prop("control")).to.equal(control);
+		expect(wrapper.prop("propertyId")).to.equal(propertyId);
+		expect(wrapper.prop("controller")).to.equal(controller);
+	});
+
+	it("toggle should render correctly", () => {
+		const wrapper = mount(
+			<Toggle
+				store={controller.getStore()}
+				control={control}
+				controller={controller}
+				propertyId={propertyId}
+			/>
+		);
+		const toggleWrapper2 = wrapper.find("span.bx--toggle__switch");
```

Comment:
These don't really test anything.  It's best to stay away from carbon specific class when not needed.  We don't need to text that carbon shows the correct classes only that the control gets rendered and we can change the value.

Testing custom labels would also be imported.